interop-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -186,13 +195,22 @@ public void onError(Throwable cause) {
     return new StreamObserver<StreamingOutputCallRequest>() {
       @Override
```

Comment:
I would rather add the boolean field `canceled` in this anonymous `StreamObserver` class than in the `ResponseDispatcher` class. (add `ResponseDispatcher#clear()` if necessary)